Enable UI for Flipper

dmarcoux opened the pull request:
This UI is only available to users with the Staff or Admin roles. It allows us to quickly manage which feature toggles are enabled/disabled without having to run commands directly on the server where our application is hosted. 

Upstream docs for the Flipper UI:
https://github.com/jnunemaker/flipper/blob/4da0b9a596465310882c368fa5292dcbbde48399/docs/ui/README.md

Once this PR is merged, this will have to be documented in our developer wiki: https://github.com/openSUSE/open-build-service/wiki/Feature-Toggles

vpereira commented on src/api/app/lib/routes_helper/role_matcher.rb:
```diff
@@ -0,0 +1,14 @@
+      current_user_login = request.session[:login]
+      current_user = current_user_login.present? ? User.find_by_login(current_user_login) : User.possibly_nobody
+
+      binding.pry
```
:eyes:

dmarcoux replied:
Yep... I still need to test some things hehehe :slightly_smiling_face:

vpereira commented on src/api/spec/services/webui_controller_service/user_checker_spec.rb:
```diff
@@ -24,25 +24,6 @@
   end
 
-  describe '#login_exists?' do
-    let(:request) { OpenStruct.new(env: { 'HTTP_X_USERNAME' => user_login }) }
```
ok, you embedded the instance methods in `if/else` inside of the `call` and you don't need to test it anymore? Wouldn't be possible to test call return in different conditions with multiple `context`?

dmarcoux replied:
I plan to rework the specs a bit. I realized this yesterday evening when seeing the coverage report.

hennevogel commented:
I would separate the refactoring and the introduction of the UI. Authentication is a very sensible code-path...

dmarcoux commented:
> I would separate the refactoring and the introduction of the UI. Authentication is a very sensible code-path...

Alright, I'll split this in two PRs. I'll rebase this PR once the refactoring is merged.

dmarcoux commented:
The refactoring of  `Webui::WebuiController#check_user` is now merged on `master`, so I rebased this PR and reworked the specs. Please have another look.

hennevogel commented on src/api/app/lib/routes_helper/role_matcher.rb:
```diff
@@ -0,0 +1,14 @@
+
+      return false unless WebuiControllerService::UserChecker.new(http_request: request, config: CONFIG).call
+
+      current_user_login = request.session[:login]
```
Hm, this isn't really set in proxy mode right?

hennevogel replied:
Ah I got tricked by `User.possibly_nobody`...
